catalog/dags/data_refresh/remap_table_constraints.py

```diff
@@ -0,0 +1,233 @@
+"""
+# Remap Table Constraints
+
+This file contains TaskGroups related to remapping the table constraints from the live media table onto the temp table to prepare it for promotion.
+
+"""
+
+import logging
+from typing import NamedTuple
+
+from airflow.decorators import task, task_group
+
+from common.sql import run_sql
+from data_refresh import queries
+
+
+logger = logging.getLogger(__name__)
+
+
+class ConstraintInfo(NamedTuple):
+    constraint_table: str
+    constraint_name: str
+    constraint_statement: str
+
+
+def fetch_all_tuples(cursor):
+    try:
+        rows = cursor.fetchall()
+        return [ConstraintInfo(row[0], row[1], row[2]) for row in rows]
+    except Exception as e:
+        raise ValueError("Unable to extract expected row data from cursor") from e
+
+
+def _is_foreign_key(_statement, table):
+    return f"REFERENCES {table}(" in _statement
+
+
+def _remap_constraint(
+    constraint_name: str,
+    constraint_table: str,
+    constraint_statement: str,
```

Comment:
These arguments directly mirror the `NamedTuple` above - even if we construct a new tuple or replace the values, would it be easier to pass around one object here and in the other places they're referenced, rather than 3 separate arguments?

Reply:
With the variable names now shortened, certainly!